test/tests/serial/mail_with_ldap.bats

```diff
@@ -1,245 +1,314 @@
-  assert_success
-  assert_output 1
+@test "spoofing (with LDAP): accepts sending as alias" {
+  _run_in_container_bash 'openssl s_client -quiet -connect 0.0.0.0:465 < /tmp/docker-mailserver-test/auth/ldap-smtp-auth-spoofed-alias.txt'
+  assert_output --partial 'End data with'
 }
 
-@test "checking dovecot: ldap mail delivery works for a different domain then the mailserver" {
-  run docker exec mail_with_ldap /bin/sh -c "sendmail -f user@external.tld some.other.user@${FQDN_LOCALHOST_B} < /tmp/docker-mailserver-test/email-templates/test-email.txt"
-  sleep 10
-  run docker exec mail_with_ldap /bin/sh -c "ls -A /var/mail/${FQDN_LOCALHOST_A}/some.other.user/new | wc -l"
-  assert_success
-  assert_output 1
-}
+@test "spoofing (with LDAP): uses senders filter" {
+  # skip introduced with #3006, changing port 25 to 465
+  # Template used has invalid AUTH: https://github.com/docker-mailserver/docker-mailserver/pull/3006#discussion_r1073321432
+  skip 'TODO: This test seems to have been broken from the start (?)'
 
-@test "checking dovecot: ldap config overwrites success" {
-  run docker exec mail_with_ldap /bin/sh -c "grep 'uris = ldap://ldap' /etc/dovecot/dovecot-ldap.conf.ext"
-  assert_success
-  run docker exec mail_with_ldap /bin/sh -c "grep 'tls = no' /etc/dovecot/dovecot-ldap.conf.ext"
-  assert_success
-  run docker exec mail_with_ldap /bin/sh -c "grep 'base = ou=people,dc=localhost,dc=localdomain' /etc/dovecot/dovecot-ldap.conf.ext"
-  assert_success
-  run docker exec mail_with_ldap /bin/sh -c "grep 'dn = cn=admin,dc=localhost,dc=localdomain' /etc/dovecot/dovecot-ldap.conf.ext"
-  assert_success
+  _run_in_container_bash 'openssl s_client -quiet -connect 0.0.0.0:465 < /tmp/docker-mailserver-test/auth/ldap-smtp-auth-spoofed-sender-with-filter-exception.txt'
+  assert_output --partial 'Sender address rejected: not owned by user'
 }
 
-@test "checking dovecot: postmaster address" {
-  run docker exec mail_with_ldap /bin/sh -c "grep 'postmaster_address = postmaster@${FQDN_LOCALHOST_A}' /etc/dovecot/conf.d/15-lda.conf"
-  assert_success
-}
+@test "saslauthd: ldap smtp authentication" {
+  # Requires ENV `PERMIT_DOCKER=container`
+  _run_in_container_bash 'nc -w 5 0.0.0.0 25 < /tmp/docker-mailserver-test/auth/sasl-ldap-smtp-auth.txt'
+  assert_output --partial 'Error: authentication not enabled'
 
-@test "checking dovecot: quota plugin is disabled" {
- run docker exec mail_with_ldap /bin/sh -c "grep '\$mail_plugins quota' /etc/dovecot/conf.d/10-mail.conf"
- assert_failure
- run docker exec mail_with_ldap /bin/sh -c "grep '\$mail_plugins imap_quota' /etc/dovecot/conf.d/20-imap.conf"
- assert_failure
- run docker exec mail_with_ldap ls /etc/dovecot/conf.d/90-quota.conf
- assert_failure
- run docker exec mail_with_ldap ls /etc/dovecot/conf.d/90-quota.conf.disab
- assert_success
-}
+  _run_in_container_bash 'openssl s_client -quiet -connect 0.0.0.0:465 < /tmp/docker-mailserver-test/auth/sasl-ldap-smtp-auth.txt'
```

Comment:
Just as a reminder for the future: `_send_email` should probably get a "ports" option and choose between `openssl s_client` and `nc` dynamically. Or, simpler, we provide a new helper `_send_mail_openssl()`.

Reply:
We could, but instead of switching between `openssl` / `nc` it might also be good to just use a CLI tool that sends mail and can support all ports?

Preferably it'd ingest a different mail template, I'd prefer we handle the auth ones with the account password in plaintext, instead of the encoded (hashed? base64?) format exchanged presently for `AUTH LOGIN`. We have an issue with that for the skipped test, and the `.ldif` files have `SSHA` hashed passwords themselves, so I've no clue what they were originally 😅

Reply:
Yeah, that might also be a good idea. But no rush, it works fine for now